From 1cc9b1423c5df591c615ef9588b6eefd81448f80 Mon Sep 17 00:00:00 2001
From: Lukáš Hejtmánek <xhejtman@ics.muni.cz>
Date: Fri, 06 Mar 2009 19:44:03 +0000
Subject: Fix another VT switch leak

The batch_bo buffer object is reallocated on enter VT, so we need to
unref it on leave vt.

Signed-off-by: Lukas Hejtmanek <xhejtman@ics.muni.cz>
---
diff --git a/src/i830_batchbuffer.c b/src/i830_batchbuffer.c
index 671e8c8..33da43e 100644
--- a/src/i830_batchbuffer.c
+++ b/src/i830_batchbuffer.c
@@ -159,6 +159,7 @@ intel_batch_teardown(ScrnInfoPtr pScrn)
 
     if (pI830->batch_ptr != NULL) {
 	dri_bo_unmap(pI830->batch_bo);
+	dri_bo_unreference(pI830->batch_bo);
 	pI830->batch_ptr = NULL;
     }
 }
--
cgit v0.8.2
